parachain/test-parachains/adder/src/wasm_validation.rs

```diff
@@ -17,12 +17,13 @@
 //! WASM validation for adder parachain.
 
 use crate::{HeadData, BlockData};
-use core::{intrinsics, panic};
+use core::panic;
+use sp_std::vec::Vec;
 use parachain::primitives::{ValidationResult, HeadData as GenericHeadData};
 use codec::{Encode, Decode};
 
 #[no_mangle]
-pub extern fn validate_block(params: *const u8, len: usize) -> u64 {
+pub extern "C" fn validate_block(params: *const u8, len: usize) -> u64 {
```

Comment:
It would be preferred to have platform independent integer width length for `len`, since this is exported.

Reply:
I guess, this won't harm, but note that this code is for wasm only and we assume that it's 32 bits only.

Reply:
It's just a beartrap in the woods, soon to be forgotten of its existence, and somebody will step into it eventually - it might not be a :bear:

Reply:
https://github.com/paritytech/polkadot/pull/1944